build-tools/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -1435,30 +1435,57 @@ private void tweakJvmOptions(Path configFileRoot) {
         Path jvmOptions = configFileRoot.resolve("jvm.options");
         try {
             String content = new String(Files.readAllBytes(jvmOptions));
-            Map<String, String> expansions = jvmOptionExpansions();
-            for (String origin : expansions.keySet()) {
-                if (content.contains(origin) == false) {
-                    throw new IOException("template property " + origin + " not found in template.");
+            Map<ReplacementKey, String> expansions = jvmOptionExpansions();
+            for (var entry : expansions.entrySet()) {
+                ReplacementKey replacement = entry.getKey();
+                String key = replacement.key();
+                if (content.contains(key) == false) {
+                    key = replacement.fallback();
+                    if (content.contains(key) == false) {
+                        throw new IOException("Template property '" + replacement + "' not found in template:\n" + content);
+                    }
                 }
-                content = content.replace(origin, expansions.get(origin));
+                content = content.replace(key, entry.getValue());
             }
             Files.write(jvmOptions, content.getBytes());
         } catch (IOException ioException) {
             throw new UncheckedIOException(ioException);
         }
     }
 
-    private Map<String, String> jvmOptionExpansions() {
-        Map<String, String> expansions = new HashMap<>();
+    private record ReplacementKey(String key, String fallback) {}
+
+    private Map<ReplacementKey, String> jvmOptionExpansions() {
+        Map<ReplacementKey, String> expansions = new HashMap<>();
         Version version = getVersion();
-        String heapDumpOrigin = getVersion().onOrAfter("6.3.0") ? "-XX:HeapDumpPath=data" : "-XX:HeapDumpPath=/heap/dump/path";
-        expansions.put(heapDumpOrigin, "-XX:HeapDumpPath=" + confPathLogs);
-        if (version.onOrAfter("6.2.0")) {
-            expansions.put("logs/gc.log", confPathLogs.resolve("gc.log").toString());
+
+        ReplacementKey heapDumpPathSub;
+        if (version.before("8.18.0") && version.onOrAfter("6.3.0")) {
```

Comment:
I don't understand these `if`s: why are they needed?

Reply:
The heap dump path argument was already previously changed. This condition maintains what it was previously, only adjusting to the logs dir (ie no heap dump path argument) on 8.18.0+ (I'll adjust this now this to 8.19.0+, though).

Reply:
I assume this is used also to launch older cluster versions (e.g. for BwC tests or upgrade; but do we need to keep 6.3? or even 7? would we ever run those using this code in main?)

Reply:
You have it right, this is for bwc. So this value we are selecting based on version is the string we expect to find in the old jvm options, which we then do a substitution on.

Reply:
OK fair enough